### ISSUE TITLE
[Suggestion] Alternative mode maglock

### ISSUE THREAD

**Issue by ZachZent** [tsunamayo/Starship-EVO#3233]
The maglocks are great, but the current ones only serve half of the situations you would need them for. A crane can lock onto a ship and move it around works fine, but if a fighter wants to attach on a ship using a maglock it would become the main entity. It is rather difficult and very inconvenient for the large ship to have the maglocks in the right place and a player who turns them on at the right time. Here are two possible ways to allow the smaller entity to latch onto the larger one while staying it “sub”

1.	Configure menu
Similar to how you would configure a turret core, this option would have a menu when the maglock is clicked allowing the player to “invert” the lock
2.	New maglock block
A simple solution with adding an “inverted” maglock block next to the current one. Maybe with a slightly different model with the gaps and the extended parts swapped as to differentiate them from each other

**Comment by cloudrambler** [comment]
An alternative method would be to provide a config option for the existing maglock block, allowing for it to either ignore targets over its mass threshold (as per current situation), or alternatively to become the movable entity if the target is larger and itself is less than the mass threshold.

**Comment by ProPeach** [comment]
A while back there were some changes discussed to maglocks that would help with the mass limit side of things, namely:

-Mass limit applies to the smaller of the two entities being locked
- multiple maglocks can be used to raise the effective limit of a lock

I hope these are still in consideration!

With regards to the issue of the ship being locked on to unable to move, imo it'd be more logical if both entities could still move, but acted as one mass. This would sidestep the issue of needing and "inverted" maglock or GUI, although I recognise this might need some changes to the physics side of things

**Comment by AlienXtream** [comment]
having 2 modes that apply depending on context/mass would be how i would set it up. if the 2 ships mass are the same (within some threshold. say within 25% of each other) then they function as one physics entity (basically as if they were one ship. so same rules for multiple cockpits/yokes apply. first in first served. a failsafe condition if more than one pilot is detected would need to be implemented) if they are significantly different masses or one is a structure then the larger of the 2 is used only. maglocks should be 2 way. i.e they can be either on the ship thats landing or on the thing its landing on. mass limit should be removed and would not be needed if using a mass based parenting. they should also be passive and default to on. much of that logic is the same as space engineers (other than the mass thing because SE uses true physics simulation whereas EVO uses a more optimized but less physically accurate method)

also, i would replace the shipyard with a "landing zone" which would just be used for the AI to know where and how to land as having both is kinda redundant.

thats my "2" cents worth on the subject :)

**Comment by tsunamayo** [comment]
Yes I want to do what @ProPeach is suggesting. Also yes @AlienXtream I agree both shipyard and maglock are a bit redundant, I will make the shipyard one evolve a bit. Maybe a AI waypoint or a thing of the like. Thx
